tests/integration/test_charm.py

```diff
@@ -120,3 +124,57 @@ async def test_get_primary_action(ops_test: OpsTest) -> None:
 
         # assert get-primary returned the right primary
         assert identified_primary == expected_primary
+
+
+async def test_exactly_one_primary_reported_by_juju(ops_test: OpsTest) -> None:
+    """Tests that there is exactly one replica set primary unit reported by juju"""
+    
+    async def get_unit_messages():
+        """Collects unit status messages"""
+        app = await app_name(ops_test)
+        unit_messages = {}
+
+        async with ops_test.fast_forward():
+            time.sleep(20)
+        
+        for unit in ops_test.model.applications[app].units:
+            unit_messages[unit.entity_id] = unit.workload_status_message
+
+        return(unit_messages)
+
+    def single_primary(unit_messages):
+        """Confirms there is only one replica set primary unit"""
+        count = 0
+        for value in unit_messages:
+            if unit_messages[value] == "Replica set primary":
+                count += 1
+
+        if count > 1:
+            assert count == 1, "Multiple replica set primary units detected"
+        elif count < 1:
+            assert count == 1, "No replica set primary units detected"
```

Comment:
Thorough, but it's enough to do:
``
```suggestion
        assert count == 1, f"juju is expected to report one primary not {count} primaries"
```